doc/init.md

```diff
@@ -65,6 +65,18 @@ reasons to make the configuration file and data directory only readable by the
 bitcoin user and group.  Access to bitcoin-cli and other bitcoind rpc clients
 can then be controlled by group membership.
 
+NOTE: When using the systemd .service file, the creation of the aforementioned
+directories and the setting of their permissions is automatically handled by
+systemd. Directories are given a permission of 710, giving the bitcoin group
+access to files under it _if_ the files themselves give permission to the
+bitcoin group to do so (e.g. when `-sysperms` is specified). This does not allow
+for the listing of files under the directory.
+
+NOTE: It is not currently possible to override `datadir` in
+`/etc/bitcoin/bitcoin.conf` with the current systemd, OpenRC, and Upstart init
+files. The command line arguments specified in the init files take precedence
+over the configurations in `/etc/bitcoin/bitcoin.conf`.
```

Comment:
At least OpenRC's init allows for setting `BITCOIND_DATADIR` to override it.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/12255#discussion_r248015409

> At least OpenRC's init allows for setting BITCOIND_DATADIR to override it.

re: http://www.erisian.com.au/bitcoin-core-dev/log-2019-01-17.html#l-365

> I see... I think I'll make it easy to override for systemd as well then. Thanks!

It would be good to mention `/etc/conf.d/bitcoind` for OpenRC, but I wouldn't put a huge amount effort into implementing a configuration mechanism for the systemd service (unless your really want to). RC scripts tend to support external configuration because can they can be pretty long and complicated. Systemd service files are (in theory) supposed to stay simple and rely on more coarse override mechanisms like `systemctl edit`.